reconcile/utils/terrascript_client.py

```diff
@@ -1536,17 +1536,72 @@ def populate_tf_resource_s3(self, resource, namespace_info):
                     'events': s3_events
                 }]
             }
-            filter_prefix = common_values.get('filter_prefix', None)
-            if filter_prefix is not None:
-                notification_values['queue'][0]['filter_prefix'] = \
-                    filter_prefix
-            filter_suffix = common_values.get('filter_suffix', None)
-            if filter_suffix is not None:
-                notification_values['queue'][0]['filter_suffix'] = \
-                    filter_suffix
+            prefix = common_values.get('filter_prefix', None)
+            if prefix is not None:
+                notification['queue'][0]['filter_prefix'] = \
+                    prefix
+            suffix = common_values.get('filter_suffix', None)
+            if suffix is not None:
+                notification['queue'][0]['filter_suffix'] = \
+                    suffix
 
             notification_tf_resource = aws_s3_bucket_notification(
-                sqs_identifier, **notification_values)
+                sqs_identifier, **notification)
+            tf_resources.append(notification_tf_resource)
+        event_notifications = common_values.get('event_notifications')
+
+        for event_notification in event_notifications:
+            destination_type = event_notification['destination_type']
+            detination_identifier = event_notification['destination']
+            event_type = event_notification['event_type']
+
+            if detination_identifier is not None:
+                identifier_is_arn = True \
+                    if detination_identifier.startswith('arn:') else False
+            if identifier_is_arn:
+                resource_name = detination_identifier.split(':')[-1]
+                resource_arn = detination_identifier
+            else:
+                resource_name = detination_identifier
+                resource_arn = '${data.aws_sns_topic.' \
```

Comment:
data.aws_sns_topic or data.aws_sqs_queue

need to be  determined by destination_type